tls/s2n_tls13_handshake.c

```diff
@@ -48,3 +48,103 @@ int s2n_tls13_keys_from_conn(struct s2n_tls13_keys *keys, struct s2n_connection
 
     return 0;
 }
+
+int s2n_tls13_select_key_and_compute_secret(struct s2n_connection *conn, struct s2n_blob *shared_secret)
+{
+    int selection = 0; /* hardcoded to the first curve */
```

Comment:
Ideally,  all this function should be doing is computing the shared secret for the server/client and it should not be selecting the curve to be used. The selection of the key happens prior to calling this function (in Server Hello) and can be obtained from the `server_ecc_params->negotiated_curve`. 

What we need here is logic to verify that there exists a  `client_ecc_params[i]->negotiated_curve` that is equal to the `server_ecc_params->negotiated_curve` and if so, use that `client_ecc_params[i]` to compute the shared secret.

Reply:
I agree that curve selection should be done at this point (also why I've added comment here). I believe we still need to figure out where the best place is to verify and select compatible curve. It also needs to work with the scenario that result in a hello retry request when no key share matches. By the time it reaches the code to compute the shared secret, ideally it doesn't have to do additional matching.  We should be follow up with the task to finalize curve selection.

Reply:
I believe the logic to verify and select compatible curve should exist in the `s2n_extensions_server_key_share_recv` for the client: 
https://github.com/awslabs/s2n/blob/c78c60ccb3f371f0d98af68995a5ed4508ca8547/tls/extensions/s2n_server_key_share.c#L98